### ISSUE TITLE
Polling mechanism doesn't work

### ISSUE THREAD

**Issue by kylebarron** [kylebarron/stata_kernel#60]
All blocks are run through DoCommandAsync. So this code isn't correctly holding the Python program:
https://github.com/kylebarron/stata_kernel/blob/49ab85460b3a52e443878352b530a0b2ee61123f/stata_kernel/stata_session.py#L299-L308

<img width="648" alt="image" src="https://user-images.githubusercontent.com/15164633/43996967-ce7a4d68-9d9c-11e8-9a3f-6313e6e08dde.png">

Thoughts:
When I move to a single log file, I could _expect_ for a `\n\n\. ` (or `\r\n\r\n\. ` for TTYs and Windows log files) in the log file stream. This is essentially the same as `pexpect` is doing for the console already.

I could even move the parsing out of what pexpect gives me and instead only use a single open log file on Windows, Linux, and Mac. And then have the exact same parsing code for all platforms.

Thoughts @mcaceresb?

**Comment by mcaceresb** [comment]
This code is working for me. Do you still get the error?

**Comment by kylebarron** [comment]
On Windows?

**Comment by kylebarron** [comment]
or Mac? This is specifically with Automation.

**Comment by mcaceresb** [comment]
So this works in the console but not the automation.

**Comment by mcaceresb** [comment]
I only have the one laptop. I don't think I've ever used python on windows tbh

**Comment by kylebarron** [comment]
Yes. I have Linux desktop, Windows VM inside Linux, and a 6-yr old Macbook that still runs great. This specifically seems to be a problem with the Stata Automation suite, because it would seem that I explicitly wait for Stata to say it's free:
https://github.com/kylebarron/stata_kernel/blob/master/stata_kernel/stata_session.py#L421-L434

But that function returns before Stata is done. Unless I coded it wrong it would seem to say that `UtilIsStataFree` returns `True` while it's sleeping.
